### PR TITLE
Update 'epic' issue template to better align with planning needs

### LINKED ISSUE #909: Update epic issue form
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

The epic form needs better fields to cover epic questions, e.g.:
- What
- Why
- User impact

### Why is this needed?

_No response_

### What should happen?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by pseudoyim**
### Description

Proposing edits and new fields in the "Epic" issue template to better capture information that will facilitate planning and prioritization efforts. Specifically, the new fields require the issue creator to clearly define the "what", "why", and "user impact" of the epic.

Resolves #909

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/epic.yml [hunk 1]
```diff
@@ -25,23 +25,75 @@ body:
...
+  - type: textarea
+    id: acceptance_criteria
+    attributes:
+      label: Acceptance criteria
+      description: >-
+        At what point can we close this epic as "completed"?
+      value: |
+        - [ ] <acceptance criterion>
+    validations:
+      required: true
```
Comment (jezdez) [inline .github/ISSUE_TEMPLATE/epic.yml:73]: That's jargon-ladden and will mean nothing to users and contributors to conda. I say we replace this with a "Goals" section instead

**Review by jezdez — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/epic.yml (hunk 1 above)
Comment (pseudoyim) [reply .github/ISSUE_TEMPLATE/epic.yml]: Done - thank you!

File: .github/ISSUE_TEMPLATE/epic.yml [hunk 2]
```diff
@@ -25,23 +25,75 @@ body:
...
-      label: Linked Issues & PRs
-      description: List all issues related to this epic.
+      label: Why?
+      description: >-
+        Why is the reported issue(s) a problem, or why is the proposed feature needed?
+        (Research and spike issues can be linked here.)
       value: |
-        - [ ] #
+        Please describe here and/or link to relevant supporting issues.
+        - [ ] <supporting issue from user, research, or spike>
```
Comment (kenodegard) [inline .github/ISSUE_TEMPLATE/epic.yml:50]:
```suggestion
        - [ ] <supporting issue from user, research, or spike>
      placeholder: Please describe here and/or link to relevant supporting issues.
```

File: .github/ISSUE_TEMPLATE/epic.yml [hunk 3]
```diff
@@ -25,23 +25,75 @@ body:
...
+      description: >-
+        Epics and issues that block or are blocked by this epic.
+      value: |
+        ### This epic is blocked by
+        - [ ] <dependency epic or issue>
+
+        ### This epic blocks
+        - [ ] <dependency epic or issue>
+    validations:
+      required: false
```
Comment (kenodegard) [inline .github/ISSUE_TEMPLATE/epic.yml:99]: better to make this two separate fields?

```suggestion
  - type: textarea
    id: blocked_by
    attributes:
      label: Blocked by
      description: >-
        Epics and issues that are blocked by this epic.
      value: |
        - [ ] <dependency epic or issue>
    validations:
      required: false
      
  - type: textarea
    id: blocking
    attributes:
      label: Blocking
      description: >-
        Epics and issues that block this epic.
      value: |
        - [ ] <dependency epic or issue>
    validations:
      required: false
```

**Review by kenodegard — COMMENTED** [review]
I think we're being inconsistent with the `placeholder` and `value` fields (see rendered form here: https://github.com/pseudoyim/infrastructure/blob/update-epic-issue-template/.github/ISSUE_TEMPLATE/epic.yml)

File: .github/ISSUE_TEMPLATE/epic.yml (hunk 2 above)
Comment (pseudoyim) [reply .github/ISSUE_TEMPLATE/epic.yml]: Thanks, @kenodegard! I'm now using `placeholder` for "Please describe..." hints, and `value` for empty checkboxes.

File: .github/ISSUE_TEMPLATE/epic.yml (hunk 3 above)
Comment (pseudoyim) [reply .github/ISSUE_TEMPLATE/epic.yml]: Thanks @kenodegard! Two separate fields def makes sense.

**Review by kenodegard — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**